src/gcp/cloudfunctions.ts

```diff
@@ -356,11 +356,16 @@ export async function updateFunction(
   cloudFunction: Omit<CloudFunction, OutputOnlyFields>
 ): Promise<Operation> {
   const endpoint = `/${cloudFunction.name}`;
+  // TODO(colerogers): remove cloudFunctionWithoutHttpsTrigger and set securityLevel in functionFromEndpoint during a breaking change release
+  const cloudFunctionWithoutHttpsTrigger = { ...cloudFunction };
+  delete cloudFunctionWithoutHttpsTrigger.httpsTrigger;
```

Comment:
This can work for now, but it won't scale if we have additional fields added to httpsTrigger. Is `prepare.ts:inferDetailsFromExisting` a good candidate here?

Reply:
Latest commit we set securityLevel in `backend.Endpoint` and use `prepare.ts:inferDetailsFromExisting` to copy the current value.